test/e2e/framework/ingress_utils.go

```diff
@@ -648,8 +674,7 @@ func (cont *GCEIngressController) isHTTPErrorCode(err error, code int) bool {
 
 // Cleanup cleans up cloud resources.
 // If del is false, it simply reports existing resources without deleting them.
-// It always deletes resources created through it's methods, like staticIP, even
-// if del is false.
+// If dle is true, it deletes resources it finds acceptable (see canDelete func).
```

Comment:
Very gross, indeed. It's also looking at the return strings of `deleteXYZ(del bool) string` to determine if resources exists. Frankly, all of it would benefit from being rebuilt.  I'm more concerned with stopping the leak of resources at the moment.